drivers/i2s/i2s_ll_stm32.c

```diff
@@ -226,21 +246,31 @@ static int i2s_stm32_configure(const struct device *dev, enum i2s_dir dir,
 
 	/* set I2S bitclock */
 	bit_clk_freq = i2s_cfg->frame_clk_freq *
-		       i2s_cfg->word_size * i2s_cfg->channels;
-
-	ret = i2s_stm32_set_clock(dev, bit_clk_freq);
-	if (ret < 0) {
-		return ret;
-	}
+		       channel_length * num_channels;
 
-	/* set I2S Master Clock */
-	if (stream->master) {
+	/* set I2S Master Clock output in the MCK pin, enabled in the DT */
+	if (stream->master && cfg->master_clk_sel) {
 		LL_I2S_EnableMasterClock(cfg->i2s);
+		/*
+		 * Compensate for the master clock dividers.
+		 * MCK = N * CK, where N:
+		 * 8 when the channel frame is 16-bit wide
+		 * 4 when the channel frame is 32-bit wide
+		 */
+		bit_clk_freq *= channel_length == 16U ? 4U * 2U : 4U;
 	} else {
 		LL_I2S_DisableMasterClock(cfg->i2s);
 	}
 
-	/* set I2S Data Format */
+	ret = i2s_stm32_set_clock(dev, bit_clk_freq);
+	if (ret < 0) {
+		return ret;
```

Comment:
with the change on i2s_stm32_set_clock,  the -EDOM is a possible return code in case of error (see above 
```
	if ((uint32_t) freq_in < bit_clk_freq) {
		LOG_ERR("The requested sample rate exceeds the source clock");
		return -EDOM;
	}
```

this _-EDOM_ could be propagated to the i2s_stm32_configure() . However this value is not listed as a return code of the  [i2s_configure](https://docs.zephyrproject.org/latest/hardware/peripherals/audio/i2s.html#i2s)

Reply:
Good point, will rework to just return `EINVAL` and that way be in line with the API documentation.